repos/system_upgrade/el7toel8/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -130,30 +132,38 @@ def _create_target_userspace_directories(target_userspace):
         )
 
 
-def gather_target_repositories(target_rhsm_info):
+def gather_target_repositories(context):
     """
-    Performs basic checks on requirements for RHSM repositories and returns the list of target repository ids to use
+    Perform basic checks on requirements for RHSM repositories and return the list of target repository ids to use
     during the upgrade.
+
+    :param context: An instance of a mounting.IsolatedActions class
+    :type context: mounting.IsolatedActions class
+    :return: List of target system repoids
+    :rtype: List(string)
     """
+    # Get the RHSM repos available in the RHEL 8 container
+    available_repos = rhsm.get_available_repo_ids(context)
+
     # FIXME: check that required repo IDs (baseos, appstream)
     # + or check that all required RHEL repo IDs are available.
     if not rhsm.skip_rhsm():
-        if not target_rhsm_info.available_repos or len(target_rhsm_info.available_repos) < 2:
+        if not available_repos or len(available_repos) < 2:
             raise StopActorExecutionError(
-                message='Cannot find required basic RHEL repositories.',
+                message='Cannot find required basic RHEL 8 repositories.',
                 details={
-                    'hint': ('It is required to have RHEL repository on the system'
+                    'hint': ('It is required to have RHEL repositories on the system'
                              ' provided by the subscription-manager. Possibly you'
                              ' are missing a valid SKU for the target system or network'
                              ' connection failed. Check whether your system is attached'
-                             ' to the valid SKU providing target repositories.')
+                             ' to a valid SKU providing RHEL 8 repositories.')
```

Comment:
just nitpick: replacing `target` wording by `RHEl 8` is little contra-productive as we will have to rewrite it in future again. Many actors could be used in future as well for another system and it is better to learn people what is "target OS" in our case than write the thing two times. But it's minor thing and definitely not blocking the merge.

Reply:
IMHO, using `target` in the message just makes a room for doubt or confusion. `RHEL 8` is specific and non-ambiguous. If we ever need to reuse this piece of code, it's trivial to update it.